daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/ElementCombinator.scala

```diff
@@ -114,7 +114,8 @@ class ElementCombinator(
         uSetVars,
         eBeforeUnparser,
         eUnparser,
-        eAfterUnparser)
+        eAfterUnparser,
+        context.ovcCompiledExpression)
```

Comment:
This is actually a good example of where something doesn't belong on the ERD, but should instead be a parameter to a parser/unparser. Because it is needed in exactly one place. 

This has the downside that elements with OVC won't be checked unless we specifically construct one for the grammar. I suppose we should demand the grammar object for every top level thing to force checking, but then only demand the parser/unparser for actual root elements.

Reply:
> I suppose we should demand the grammar object for every top level thing

Do you want me to add that as part of this pull request? Does that essentially mean doing something like
```scala
globalElementDecls.foreach { _.asRoot }
```

Also, do we want to do it for all imported/included schemas, or only top level elements in the first/main schema?

Reply:
We don't have to do that now. If you do that now, then I think we'd still have the problem where some schemas need to be modified to turn off compileAllTopLevel.

Reply:
Ok, I've created [DAFFODIL-2556](https://issues.apache.org/jira/browse/DAFFODIL-2556) to track that potential change.